Sources/SpelledPitch/PitchSpeller/Wetherfield/FlowNetwork.swift

```diff
@@ -35,44 +35,55 @@ extension FlowNetwork {
 }
 
 extension WeightedDirectedGraph where Weight: Comparable {
-    // TODO: Make throw
-    mutating func reduceFlow(through edge: Edge, by amount: Weight) {
-        updateEdge(edge) { weight in weight - amount }
-    }
 
-    /// Removes the given edge if its weight is `0`. This happens after an edge, which has the
-    /// minimum flow of an augmenting path, is reduced by the minimum flow (which is its previous
-    /// value).
-    mutating func removeEdgeIfFlowless(_ edge: Edge) {
-        if weight(edge) == 0 {
-            removeEdge(edge)
+    /// Signals that modifications to the graph are necessary after a change to a certain edge
+    enum EdgeIs: Error {
+        case absent
+        case saturated
+        case oversaturated
+    }
+    
+    /// Increases the flow through the given `edge` by `amount`. Throws `edgeAbsent` if `edge`
+    /// is not found in the network's underlying graph, and `capacityExceeded` if the amount
+    /// being added exceeds the remaining capacity of the edge.
+    mutating func increaseFlow(through edge: Edge, by amount: Weight) throws {
+        guard let capacity = weight(edge) else { throw EdgeIs.absent }
+        if amount > capacity {
+            throw EdgeIs.oversaturated
+        } else if amount == capacity {
+            throw EdgeIs.saturated
+        } else {
+            updateEdge(edge) { weight in weight - amount }
         }
     }
 
-    /// Inserts an edge in the opposite direction of the given `edge` with the minimum flow
-    mutating func updateBackEdge(_ edge: Edge, by minimumFlow: Weight) {
+    /// Inserts an edge in the opposite direction of the given `edge` with the minimum capacity
+    mutating func enableBackFlow(through edge: Edge, equalTo amount: Weight) {
         let reversedEdge = edge.swapped
         if contains(reversedEdge) {
-            updateEdge(reversedEdge) { capacity in capacity + minimumFlow }
+            updateEdge(reversedEdge) { capacity in capacity + amount }
         } else {
-            insertEdge(reversedEdge, weight: minimumFlow)
+            insertEdge(reversedEdge, weight: amount)
         }
     }
 
-    /// Reduces the flow of the given `edge` by the given `minimumFlow`. If the new flow through
-    /// the `edge` is now `0`, removes the `edge` from the network. Updates the reverse of the given
-    /// `edge` by the given `minimumFlow`.
-    mutating func pushFlow(through edge: Edge, by minimumFlow: Weight) {
-        reduceFlow(through: edge, by: minimumFlow)
-        removeEdgeIfFlowless(edge)
-        updateBackEdge(edge, by: minimumFlow)
+    /// Increases the flow through the given `edge` by the given `minimumCapacity`. If `edge` is
+    /// saturated (its flow equals its capacity), removes the `edge` from the network. Increases
+    /// the capacity for flow in the opposite direction (reversed edge) by `minimumCapacity`.
+    mutating func pushFlow(through edge: Edge, equalTo amount: Weight) {
```

Comment:
If instead of throwing (see above), you could switch directly over the result of `increaseFlow(through:by:)`:

```Swift
switch increaseFlow(through: edge, by: amount) {
    case .isSaturated:
        ...
    ...
}
```

Reply:
This is elegant. Like it.